src/Array.mo

```diff
@@ -615,13 +617,12 @@ module {
     var outer = 0;
     var inner = 0;
     Prim.Array_tabulate<X>(flatSize, func _ {
-      let subArray = arrays[outer];
-      let element = subArray[inner];
-      inner += 1;
-      if (inner == subArray.size()) {
+      while (inner == arrays[outer].size()) {
```

Comment:
Nice catch! So flatten didn't work for empty inner arrays.
I wonder whether we should proptest the standard library instead of (or in addition to) relying on manual test case construction. A basic proptest would likely uncover this bug on the first run.

Reply:
Well I should say I'm catching my own bug from #425.

But yeah, agreed that property based testing (especially in library development contexts) would be very nice